### PR TITLE
Feature/52 introduce topics

### LINKED ISSUE #52: Introduce Topics
## Background
The list of decisions made (make from this section in README as best practise proposals):
- The documentation issues should be present in the same repository as the ones used for development.
  - An isolated repository will be created. **Example:** aqueduct-documentation 
- Set of labels to identify the type of issue: `DocumentedFeature,` `DocumentedUserStory,` `DocumentedRequirement`
  - The issue will differ according to the description format. (e.g., User Story can define a list of related Features)
- Set of labels representing one Topic
  - {Name}Topic, where Topic part is required for later recognition.

## Feature
- Add new input control to enable Topic grouping in the output.
- Add the logic to group Features to stand-alone tables in main _index.page.

### PR DESCRIPTION
**Pull request by MobiTikula**
Introduce topics into the project
- Add new input control to enable Topic grouping in the output.
- Add the logic to group Features to stand-alone tables in main _index.page.
- Update the README.md file with new feature
- Update yaml file for CI run
- Check the correct behavior at mdoc viewer page

Release Notes:
- New user feature, that provides an option of grouping issues by Topics.

Closes #52

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -601,13 +616,40 @@ This feature allows you to generate structured output for the living documentati
...
       |--repo 1
          |-- issue md page 1
          |-- _index.md
       |--repo 2
+          ...
+      |-- _index.md
+    |- _index.md
+    ```
+
+### Grouped Output by Issue Topics
```
Comment (miroslavpojer) [inline README.md:630]:
```suggestion
### Grouped Output by Topics
```

File: README.md [hunk 2]
```diff
@@ -601,13 +616,40 @@ This feature allows you to generate structured output for the living documentati
...
+
+The feature allows you to generate output grouped by issue topics. This feature is useful when you want to group issues by specific topics or themes.
```
Comment (miroslavpojer) [inline README.md:632]:
```suggestion
The feature allows you to generate output grouped by topics. This feature is useful when you want to group tickets by specific topics or themes.
```

File: README.md [hunk 3]
```diff
@@ -601,13 +616,40 @@ This feature allows you to generate structured output for the living documentati
...
+
+To gain a better understanding of the term "Topic", refer to the [Labels](#labels) section.
+
+- **Default Behavior**: By default, the action generates all the documentation in a single directory.
+- **Non-default Example**: Use the grouped output feature to organize the generated documentation by issue topics.
```
Comment (miroslavpojer) [inline README.md:637]:
```suggestion
- **Non-default Example**: Use the grouped output feature to organize the generated documentation by topics.
```

File: living_documentation_generator/model/consolidated_issue.py [hunk 4]
```diff
@@ -161,3 +168,38 @@ def generate_page_filename(self) -> str:
...
+
+        # If structured output is enabled, create a directory path based on the repository
+        if ActionInputs.get_is_structured_output_enabled() and self.repository_id:
+            organization_name, repository_name = self.repository_id.split("/")
+            output_path = os.path.join(output_path, organization_name, repository_name)
+
+        # If grouping by topics is enabled, create a directory path based on the issue topic
+        if ActionInputs.get_is_grouping_by_topics_enabled():
+            # Extract labels from the issue table
+            labels = re.findall(r"\| Labels \| (.*?) \|", issue_table)
```
Comment (miroslavpojer) [inline living_documentation_generator/model/consolidated_issue.py:189]: I would add check & log when more than one topic label will be detected.

File: living_documentation_generator/model/consolidated_issue.py [hunk 5]
```diff
@@ -161,3 +168,38 @@ def generate_page_filename(self) -> str:
...
+
+            # Check for labels ending with "Topic" and generate a directory path based on it
+            for label in labels:
+                if label.endswith("Topic"):
+                    self.__topic = label
+                    topic_path = os.path.join(output_path, label)
+                    return topic_path
+
+            # If no label ends with "Topic", create a "noTopic" issue directory path
+            self.__topic = "noTopic"
```
Comment (miroslavpojer) [inline living_documentation_generator/model/consolidated_issue.py:201]:
```suggestion
            self.__topic = "NoTopic"
```

File: living_documentation_generator/generator.py [hunk 6]
```diff
@@ -290,61 +291,42 @@ def _generate_markdown_pages(self, issues: dict[str, ConsolidatedIssue]) -> None
...
         # Generate a markdown page for every issue
         for consolidated_issue in issues.values():
             self._generate_md_issue_page(issue_page_detail_template, consolidated_issue)
         logger.info("Markdown page generation - generated `%s` issue pages.", len(issues))
 
-        # Generate an index page with a summary table about all issues
-        if ActionInputs.get_is_structured_output_enabled():
-            self._generate_structured_index_pages(index_repo_level_template, index_org_level_template, issues)
+        # Generate all structure of the index pages
+        if is_structured_output:
```
Comment (miroslavpojer) [inline living_documentation_generator/generator.py:314]: This if-elif is strange as both can happen in same time.

**Review by miroslavpojer — COMMENTED** [review]
- [ ] pulled
- [ ] local run done
- [ ] pylint checked
- [ ] unit tests ran

I am missing unit tests for newly added methods in test suites which have been already implemented.
I see problem with current Outpus logic solution It will be addressed in following issue.

File: README.md (hunk 2 above)
Comment (MobiTikula) [reply README.md]: See commit: [f62aa4c](https://github.com/AbsaOSS/living-doc-generator/pull/63/commits/f62aa4ca4606f6e42dc9651590ca584e306e3f16)

File: README.md (hunk 3 above)
Comment (MobiTikula) [reply README.md]: Changes added: [f62aa4c](https://github.com/AbsaOSS/living-doc-generator/pull/63/commits/f62aa4ca4606f6e42dc9651590ca584e306e3f16).

File: living_documentation_generator/model/consolidated_issue.py (hunk 5 above)
Comment (MobiTikula) [reply living_documentation_generator/model/consolidated_issue.py]: Suggestion implemented: [b6088b4](https://github.com/AbsaOSS/living-doc-generator/pull/63/commits/b6088b4c6cdc910837caa5ce05ebd8a465a5d338).

File: living_documentation_generator/model/consolidated_issue.py (hunk 4 above)
Comment (MobiTikula) [reply living_documentation_generator/model/consolidated_issue.py:192]: [b6088b4](https://github.com/AbsaOSS/living-doc-generator/pull/63/commits/b6088b4c6cdc910837caa5ce05ebd8a465a5d338) I also added a logic that handles more than one Topic within one isssue in commit: [ecaffe0](https://github.com/AbsaOSS/living-doc-generator/pull/63/commits/ecaffe065a29c8fa8b513a4a98a1007c88af45b8)

**Comment by MobiTikula** [comment]
I did add a unit test for new logic, see commit: [dadffcc](https://github.com/AbsaOSS/living-doc-generator/pull/63/commits/dadffcca7e64eeb3c4168be38a99a649e58b6f4c).

File: living_documentation_generator/generator.py (hunk 6 above)
Comment (miroslavpojer) [reply living_documentation_generator/generator.py:317]: I am not able to provide a better proposal too.
Let's fix it in a dedicated issue.

**Review by miroslavpojer — APPROVED** [review]

**Pull request merged**